src/redux/reducers.js

```diff
@@ -0,0 +1,94 @@
+import { combineReducers } from 'redux';
+import { createSelector } from 'reselect';
+import Inmutable from 'seamless-immutable';
+
+import { actions } from './actions';
+
+const defaultState = Inmutable({
+  wBooks: {
+    filterType: '',
+    filter: '',
+    books: [],
+    filteredBooks: []
+  },
+  isFetching: false,
+  bookDetail: null
+});
+
+const search = (text, filter) => (
+  text.toLowerCase().indexOf(filter.toLowerCase()) > -1
+);
+
+const booksSelector = state => state.books;
+const filterTypeSelector = state => state.filterType;
+const filterSelector = state => state.filter;
+
+const filteredBooksSelector = createSelector(
+  booksSelector,
+  filterTypeSelector,
+  filterSelector,
+  (books, filterType, filter) => books.filter(book => search(book[filterType], filter))
+);
+
+function wBooks(state = defaultState.wBooks, action) {
+  let nextState = state;
+  switch (action.type) {
+    case actions.SET_FILTER_TYPE:
+      nextState = nextState.merge({ filterType: action.payload.filterType });
+      break;
+
+    case actions.SET_FILTER:
+      nextState = nextState.merge({ filter: action.payload.filter });
+      break;
+
+    case actions.RECEIVE_BOOKS:
+      nextState = nextState.merge({ books: action.payload.books });
+      break;
+
+    default:
+      return state;
+  }
+
+  if (nextState.filterType && nextState.filter && nextState.books.length > 0) {
```

Comment:
At first glance this is not going to be executed, since `action.type -> default` case will return the `state`.
You may extract this as another action / reducer case. After dispatching a `SET_FILTER_TYPE`, `SET_FILTER` or `RECEIVE_BOOKS`, these can dispatch a `FILTER_BOOKS` action by example.

Would you like to, in `actions.js`, do something like:
```
export function setFilterType(filterType) {
  return (dispatch, getState) => {
     dispatch({ type: actions.SET_FILTER_TYPE,  payload: { filterType });
     const nextState = getState().wBooks;
     if (nextState.filterType && nextState.filter && nextState.books.length > 0) {
       dispatch(actions.filterBooks(nextState.filterType, nextState.filter));
     }
  };
}
```
?
Or even calculate here the filter result using `filteredBooksSelector`. I'll attach another comment about this within this PR. Then, let the reducer use a `case FILTER_BOOKS` and modify the state 🙂